Add Mechanical cases of PLAID Benchmarks.

Samy926i opened the pull request.

Samy926i committed:
add mechanical task cases

Samy926i committed:
update readme mechanical task cases

Samy926i committed:
second update readme mechanical task cases

giovannicatalani commented on 2D_ElastoPlastoDynamics_task/pre_process_dataset_sdf.py:
the datasets file is missing (wither add that or add the load dataset function)
same for all the cases.

Samy926i replied:
Thank you for your recommendations. However, I didn’t understand what "the datasets file" and "the load dataset function" refer to. Can I keep a pre_dataset.py file in each case to perform preprocessing steps, or should I integrate it into the dataset.py script?

giovannicatalani replied:
My bad, I just realized datasets is the hugging face library. You can ignore my comment on that. Yes you can keep the pre_dataset.py similar to what is done in the 2D_profile folder.

giovannicatalani reviewed (changes requested):
Thanks for the PR. The code looks good in general, however some changes need to be done before it can be merged.

-Some functions are missing like load dataset (should be in datasets which is missing) for all three cases.
- The folder cases should have the same strucutre as the already existing ones. All the model, utils training conditioning should be common for all the cases and are contained in the src folder. In each folder case  it should be contained a dataset.py containing the data class and data_loader, the training files and configs but no additional utils folder.
I suggest you don't change the utlis_training.py. If you use a different version of the model (like in Tensile2D the MoE).i suggest it you call it with another name, add the loading option in load_models.py accordingly.
-  I see that sowwhere you changed the function graph_outer_step.py in utils_Training.py (adding a batch index which is not used), pleae correct that.
- Test the code before committing so that it runs  fine and all dependencies are included.

giovannicatalani commented on 2D_MultiScHypEl_task/utils/utils_training.py:
```diff
@@ -0,0 +1,214 @@
+    inner_steps,
+    inner_lr,
+    num_epoch,
+    num_batch,
```
what is num_batch, it seems like it is not used.

giovannicatalani commented on 2D_MultiScHypEl_task/utils/utils_training.py:
```diff
@@ -0,0 +1,214 @@
+    inner_steps,
+    inner_lr,
+    num_epoch,
+    num_batch,
```
num batch?

giovannicatalani commented on 2D_MultiScHypEl_task/train_sdf.py:
```diff
@@ -0,0 +1,248 @@
+                    cfg.optim.inner_steps,
+                    alpha_in,
+                    num_epoch = 100000,
+                    num_batch = 80,
```
avoid numbatch and num_Epochs. better not change the functions in utils_training.py

Samy926i committed:
Modification from Giovanni recommandation

Samy926i committed:
Modification from Giovanni recommandation 2

Samy926i commented:
I have made the necessary modifications to the code following your recommendations.
For the slightly different model I used for the 2D tensile case, I renamed it model_MOE.py and placed it in the Tensile2d_task directory. I also updated the load_model.py script to ensure it loads the model correctly, as I preferred not to modify your src/model.py script by including model_MOE.py directly within it. However, please let me know if you would prefer a different solution. I remain open to any further suggestions you may have.

giovannicatalani reviewed (approved):
Thanks a lot Samy, excellent job! I will merge and correct the small details.

casenave commented:
Thanks a lot to @Samy926i :-)
@giovannicatalani the 6 tasks are linked in the plaid repo here: https://github.com/PLAID-lib/plaid/tree/main/benchmarks/MARIO
